Tomes/TomeArray.cs

```diff
@@ -53,51 +53,78 @@ public class TomeArray : JArray {
+				// If replace was successfuly move over event handlers and call new onChanged handler
+				// The instance in which replace would not be successful, is when the old and new values are the same
+				onChanged -= EmitToParents;
+				onChanged += newTomeArray.onChanged;
+				newTomeArray.onChanged = onChanged;
+				newTomeArray.onDestroy = onDestroy;
+				onAdd -= EmitChanged;
+				onAdd += newTomeArray.onAdd;
+				newTomeArray.onAdd = onAdd;
+				onDel -= EmitChanged;
+				onDel += newTomeArray.onDel;
+				newTomeArray.onDel = onDel;
+
+				if (newTomeArray.onChanged != null) {
+					newTomeArray.onChanged.Invoke(null);
+				}
+			} else {
+				// Otherwise call original onChanged handler
+				if (onChanged != null) {
+					onChanged.Invoke(null);
+				}
 			}
 			break;
 		case JTokenType.Object:
 			TomeObject newTomeObject = new TomeObject((JObject)newValue, root);
 			this.Replace(newTomeObject);
 
-			onChanged -= EmitToParents;
-			onChanged += newTomeObject.onChanged;
-			newTomeObject.onChanged = onChanged;
-			newTomeObject.onDestroy = onDestroy;
-			onAdd -= EmitChanged;
-			onAdd += newTomeObject.onAdd;
-			newTomeObject.onAdd = onAdd;
-			onDel -= EmitChanged;
-			onDel += newTomeObject.onDel;
-			newTomeObject.onDel = onDel;
-
-			if (newTomeObject.onChanged != null) {
-				newTomeObject.onChanged.Invoke(null);
+			if (this.Parent == null) {
+				// If replace was successfuly move over event handlers and call new onChanged handler
+				// The instance in which replace would not be successful, is when the old and new values are the same
+				onChanged -= EmitToParents;
+				onChanged += newTomeObject.onChanged;
+				newTomeObject.onChanged = onChanged;
+				newTomeObject.onDestroy = onDestroy;
+				onAdd -= EmitChanged;
+				onAdd += newTomeObject.onAdd;
+				newTomeObject.onAdd = onAdd;
+				onDel -= EmitChanged;
+				onDel += newTomeObject.onDel;
+				newTomeObject.onDel = onDel;
+
+				if (newTomeObject.onChanged != null) {
+					newTomeObject.onChanged.Invoke(null);
+				}
+			} else {
+				// Otherwise call original onChanged handler
+				if (onChanged != null) {
+					onChanged.Invoke(null);
+				}
 			}
 			break;
 		default:
 			TomeValue newTomeValue = new TomeValue((JValue)newValue, root);
 			this.Replace(newTomeValue);
 
-			onChanged -= EmitToParents;
-			onChanged += newTomeValue.onChanged;
-			newTomeValue.onChanged = onChanged;
-			newTomeValue.onDestroy = onDestroy;
-
-			if (newTomeValue.onChanged != null) {
-				newTomeValue.onChanged.Invoke(null);
+			if (this.Parent == null) {
+				// If replace was successfuly move over event handlers and call new onChanged handler
+				// The instance in which replace would not be successful, is when the old and new values are the same
+				onChanged -= EmitToParents;
+				onChanged += newTomeValue.onChanged;
+				newTomeValue.onChanged = onChanged;
+				newTomeValue.onDestroy = onDestroy;
+
+				if (newTomeValue.onChanged != null) {
+					newTomeValue.onChanged.Invoke(null);
```

Comment:
I'm just a bit confused about the use here, why are we sending out null in this case?